### ISSUE TITLE
[prefer-self-closing-tags] HTML files unrelated to Angular components should be ignored

### ISSUE THREAD

**Issue by json-derulo** [angular-eslint/angular-eslint#1471]
**Description and reproduction of the issue**

In context of Angular templates, this rule makes sense, because Angular will take care of correctly parsing the element to the DOM. However if there is an HTML file in an Angular project which is not a template of an Angular component, such as `index.html`, this rule does not make sense. Here invalid HTML would be produced. As per [spec](https://html.spec.whatwg.org/multipage/syntax.html#syntax-elements) custom elements are not allowed to be self-closing. Browsers can handle it and will probably render the application correctly, nevertheless no error should be reported.

```JSON
{
  "rules": {
    "@angular-eslint/template/prefer-self-closing-tags": ["error"]
  }
}
```

**index.html**

```HTML
<!doctype html>
<html>
<body>
  <app-root></app-root>
</body>
</html>
```

**Versions**

| package                                   | version |
| ----------------------------------        | ------- |
| `@angular-eslint/eslint-plugin-template`  | `16.1.0` |
| `@angular-eslint/template-parser`         | `16.1.0` |
| `@typescript-eslint/parser`               | `6.3.0` |
| `ESLint`                                  | `8.46.0` |
| `node`                                    | `18.16.0` |

```sh
# Please run `npx ng version` in your project and paste the full output here:
Angular CLI: 16.1.8
Node: 18.16.0
Package Manager: npm 9.8.1
OS: darwin x64

Angular: 16.1.8
... animations, cli, common, compiler, compiler-cli, core, forms
... platform-browser, platform-browser-dynamic, router

Package                         Version
---------------------------------------------------------
@angular-devkit/architect       0.1601.8
@angular-devkit/build-angular   16.1.8
@angular-devkit/core            16.1.8
@angular-devkit/schematics      16.1.8
@angular/cdk                    16.1.7
@angular/material               16.1.7
@schematics/angular             16.1.8
ng-packagr                      16.1.0
rxjs                            7.8.1
typescript                      5.1.6
zone.js                         0.13.1
```

- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest supported version of the packages and checked my `ng version` output per the instructions given here.

**Comment by JamesHenry** [comment]
You can also use an eslint-ignore comment within HTML files for specific rule violations you want to ignore
